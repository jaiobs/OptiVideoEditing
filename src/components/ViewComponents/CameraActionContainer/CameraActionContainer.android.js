import React, {Component} from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
  Dimensions,
} from 'react-native';
import CameraCaptureAnimator from '../../../components/ViewComponents/CameraCaptureAnimator/CameraCaptureAnimator';
import {Images} from '../../../res';

export default class CameraActionContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentSpeed: 'Normal',
      width: 0,
    };
  }
  componentDidUpdate() {
    if (this.state.width !== Dimensions.get('window').width) {
      this.updateState();
    }
  }
  updateState() {
    this.setState({width: Dimensions.get('window').width});
  }

  renderPortraitContainer() {
    return (
      <View style={[portraitStyles.container]}>
        <View style={portraitStyles.captureLayoutPortrait}>
          <View
            style={[
              portraitStyles.gallery_icon,
              {
                width: this.state.width / 2 + 45,
              },
            ]}>
            <View style={portraitStyles.empty_view} />
            <View style={portraitStyles.shadowImage}>
              {!this.props.isRecording && !this.props.hideSilhouteIcon && (
                <TouchableOpacity
                  onPress={() => {
                    this.props.onShadowImagePressed();
                  }}
                  style={portraitStyles.gallery_button}>
                  <Image
                    style={portraitStyles.pickFromGallery}
                    source={Images.shadow_icon}
                  />
                </TouchableOpacity>
              )}
            </View>
            <View style={{width: this.state.width / 4}}>
              {this.props.HidesCameraButton && (
                <CameraCaptureAnimator
                  ref={(refs) => (this.cameraCaptureAnimator = refs)}
                  orientation={this.props.orientationCheck}
                  capturePressed={() => {
                    this.props.capturePressed();
                  }}
                  captureReleased={() => {
                    this.props.captureReleased();
                  }}
                  capturePhoto={() => {
                    this.props.capturePhoto();
                  }}
                />
              )}
            </View>
          </View>
          {!this.props.isRecording && (
            <View
              style={[
                portraitStyles.icon_view,
                {
                  width: this.state.width / 2,
                },
              ]}>
              {this.props.galleryPickerVisibility && (
                <TouchableOpacity
                  onPress={() => {
                    this.props.onGalleryPickerPressed();
                  }}
                  style={portraitStyles.gallery_button}>
                  <Image
                    style={portraitStyles.pickFromGallery}
                    source={Images.pick_gallery}
                  />
                </TouchableOpacity>
              )}

              {!this.props.galleryPickerVisibility && (
                <TouchableOpacity
                  onPress={() => {
                    this.props.onDeleteLastVideoPressed();
                  }}
                  style={portraitStyles.delete_button}>
                  <Image
                    style={portraitStyles.deleteLastIcon}
                    source={Images.delete_last}
                  />
                </TouchableOpacity>
              )}

              {!this.props.galleryPickerVisibility &&
                this.props.doneCaptureVisibility && (
                  <TouchableOpacity
                    onPress={() => {
                      this.props.onVideoFinishPressed();
                    }}
                    style={portraitStyles.video_view}>
                    <Image
                      style={portraitStyles.nextVideoIcon}
                      source={Images.next_video_icon}
                    />
                  </TouchableOpacity>
                )}
            </View>
          )}
        </View>
      </View>
    );
  }

  renderLandscapeContainer() {
    return (
      <View style={landscapeStyles.container}>
        <View style={landscapeStyles.empty_view} />
        <View style={landscapeStyles.shadow_image}>
          {!this.props.isRecording && !this.props.hideSilhouteIcon && (
            <TouchableOpacity
              onPress={() => {
                this.props.onShadowImagePressed();
              }}
              style={portraitStyles.gallery_button}>
              <Image
                style={portraitStyles.pickFromGallery}
                source={Images.shadow_icon}
              />
            </TouchableOpacity>
          )}
        </View>
        <View>
          <CameraCaptureAnimator
            ref={(refs) => (this.cameraCaptureAnimator = refs)}
            orientation={this.props.orientationCheck}
            capturePressed={() => {
              this.props.capturePressed();
            }}
            captureReleased={() => {
              this.props.captureReleased();
            }}
            capturePhoto={() => {
              this.props.capturePhoto();
            }}
          />
        </View>
        <View style={landscapeStyles.empty_view}>
          {!this.props.isRecording && (
            <View style={landscapeStyles.camera_menus}>
              {this.props.galleryPickerVisibility && (
                <TouchableOpacity
                  onPress={() => {
                    this.props.onGalleryPickerPressed();
                  }}>
                  <Image
                    style={portraitStyles.pickFromGallery}
                    source={Images.pick_gallery}
                  />
                </TouchableOpacity>
              )}

              {!this.props.galleryPickerVisibility && (
                <TouchableOpacity
                  onPress={() => {
                    this.props.onDeleteLastVideoPressed();
                  }}>
                  <Image
                    style={portraitStyles.deleteLastIcon}
                    source={Images.delete_last}
                  />
                </TouchableOpacity>
              )}

              {!this.props.galleryPickerVisibility &&
                this.props.doneCaptureVisibility && (
                  <TouchableOpacity
                    onPress={() => {
                      this.props.onVideoFinishPressed();
                    }}>
                    <Image
                      style={portraitStyles.nextVideoIcon}
                      source={Images.next_video_icon}
                    />
                  </TouchableOpacity>
                )}
            </View>
          )}
        </View>
      </View>
    );
  }

  render() {
    if (this.props.orientation === 'portrait') {
      return this.renderPortraitContainer();
    } else {
      return this.renderLandscapeContainer();
    }
  }
}

const landscapeStyles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignSelf: 'flex-end',
    height: '100%',
    position: 'absolute',
    backgroundColor: 'transparent',
    marginTop: 10,
    zIndex: 2,
  },
  captureLayoutLandscape: {
    flex: 1,
    height: '100%',
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
  },
  pickFromGallery: {
    height: 24,
    width: 24,
  },
  empty_view: {
    flex: 1,
  },
  nextVideoIcon: {
    height: 28,
    width: 28,
  },
  shadow_image: {
    height: 40,
    alignItems: 'center',
    justifyContent: 'center',
  },
  camera_menus: {
    justifyContent: 'space-between',
    alignItems: 'center',
    alignSelf: 'center',
    flex: 1,
    paddingTop: 8,
    paddingBottom: 26,
  },
  deleteLastIcon: {
    height: 32,
    width: 32,
  },
});

const portraitStyles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    bottom: 0,
    position: 'absolute',
    paddingTop: 10,
    paddingBottom: 10,
  },
  captureLayoutPortrait: {
    marginTop: 8,
    marginBottom: 8,
    flex: 1,
    width: '100%',
    flexDirection: 'row',
  },
  pickFromGallery: {
    height: 24,
    width: 24,
  },
  empty_view: {
    flex: 1,
  },
  shadowImage: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: 40,
    paddingRight: 5,
  },
  gallery_button: {
    marginLeft: 4,
  },
  delete_button: {
    marginLeft: 4,
  },
  video_view: {
    marginRight: 10,
  },
  icon_view: {
    flex: 1,
    height: '100%',
    alignSelf: 'flex-end',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  nextVideoIcon: {
    height: 28,
    width: 28,
  },
  deleteLastIcon: {
    height: 32,
    width: 32,
  },
  gallery_icon: {
    alignSelf: 'flex-end',
    alignItems: 'flex-end',
    flexDirection: 'row',
  },
  speedIndicatorText: {
    color: 'white',
    padding: 10,
    fontSize: 14,
  },
});
