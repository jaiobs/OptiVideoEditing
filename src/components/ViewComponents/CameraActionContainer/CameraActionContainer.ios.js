import {
  Dimensions,
  Image,
  StyleSheet,
  TouchableOpacity,
  View,
} from "react-native";
import React, { Component } from "react";

import CameraCaptureAnimator from "@litpic/react-native-litpic-camera-module/src/components/ViewComponents/CameraCaptureAnimator/CameraCaptureAnimator";
import CameraSpeedControls from "@litpic/react-native-litpic-camera-module/src/components/CameraSpeedControls/CameraSpeedControls";
import { Images } from "@litpic/react-native-litpic-camera-module/src/res";
import { utils } from "../../../utils";

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

export default class CameraActionContainer extends Component {
  constructor(props) {
    super(props);
    this.cameraCaptureAnimator = React.createRef();
  }

  timerToRecordVideo() {
    this.cameraCaptureAnimator.current.onPressIn();
  }

  stopToRecordVideo() {
    this.cameraCaptureAnimator.current.onPressOut();
  }

  renderPortraitOptions() {
    if (this.props.galleryPickerVisibility) {
      return (
        <View style={{ flex: 1, flexDirection: "row" }}>
          <View
            style={{
              flex: 1,
              alignItems: "flex-start",
              justifyContent: "center",
            }}
          >
            <TouchableOpacity
              onPress={() => {
                this.props.onGalleryPickerPressed();
              }}
            >
              <Image
                style={portraitStyles.pickFromGallery}
                source={Images.pick_gallery}
              />
            </TouchableOpacity>
          </View>

          <View
            style={{
              flex: 1,
              alignItems: "center",
              justifyContent: "center",
            }}
          ></View>
        </View>
      );
    } else {
      return (
        <View style={{ flex: 1, flexDirection: "row" }}>
          <View
            style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
          >
            <TouchableOpacity
              onPress={() => {
                this.props.onDeleteLastVideoPressed();
              }}
            >
              <Image
                style={portraitStyles.deleteLastIcon}
                source={Images.delete_last}
              />
            </TouchableOpacity>
          </View>
          <View
            style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
          >
            <TouchableOpacity
              onPress={() => {
                this.props.onVideoFinishPressed();
              }}
            >
              <Image
                style={portraitStyles.nextVideoIcon}
                source={Images.next_video_icon}
              />
            </TouchableOpacity>
          </View>
        </View>
      );
    }
  }

  renderIOSPortraitContainer() {
    return (
      <View style={portraitStyles.container}>
        {this.props.showSpeedView && (
          <CameraSpeedControls
            speedIndex={this.props.speedIndex}
            speedMode={this.props.speedMode}
            showSpeedView={this.props.showSpeedView}
            visibility={this.props.controlVisibility}
            orientation={this.props.orientation}
            speedDidChange={(speed) => {
              this.props.onVideoModeChanged(speed);
            }}
          />
        )}
        <View
          style={[
            portraitStyles.captureLayoutPortrait,
            { flexDirection: "row" },
          ]}
        >
          <View
            style={{ flex: 1, justifyContent: "center", alignItems: "center" }}
          >
            {!this.props.isRecording && !this.props.galleryPickerVisibility && (
              <TouchableOpacity
                style={{ marginLeft: 15 }}
                onPress={() => {
                  this.props.onPressSilhoutteMode();
                }}
              >
                <Image
                  style={[
                    portraitStyles.pickFromGallery,
                    { height: 35, width: 35, resizeMode: "contain" },
                  ]}
                  source={
                    this.props.silhoutteSelected
                      ? Images.shadow_icon_selected
                      : Images.shadow_icon
                  }
                />
              </TouchableOpacity>
            )}
          </View>
          <View style={{ flex: 1 }}>
            <CameraCaptureAnimator
              ref={this.cameraCaptureAnimator}
              flash={this.props.flash}
              type={this.props.type}
              isVisible={!this.props.isVisible}
              orientation={this.props.orientationCheck}
              disableButton={this.props.disableButton}
              capturePressed={() => {
                this.props.capturePressed();
              }}
              captureReleased={() => {
                this.props.captureReleased();
              }}
              capturePhoto={() => {
                this.props.capturePhoto();
              }}
            />
          </View>
          {!this.props.isRecording ? (
            this.renderPortraitOptions()
          ) : (
            <View style={{ flex: 1, flexDirection: "row" }} />
          )}
        </View>
      </View>
    );
  }

  renderLandscapeOptions() {
    if (this.props.galleryPickerVisibility) {
      return (
        <View style={{ flex: 1, flexDirection: "column" }}>
          <View
            style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
          >
            <TouchableOpacity
              onPress={() => {
                this.props.onGalleryPickerPressed();
              }}
            >
              <Image
                style={portraitStyles.pickFromGallery}
                source={Images.pick_gallery}
              />
            </TouchableOpacity>
          </View>
          <View
            style={{
              flex: 1,
              alignItems: "center",
              justifyContent: "center",
            }}
          ></View>
        </View>
      );
    } else {
      return (
        <View style={{ flex: 1, flexDirection: "column" }}>
          <View
            style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
          >
            <TouchableOpacity
              onPress={() => {
                this.props.onDeleteLastVideoPressed();
              }}
            >
              <Image
                style={portraitStyles.deleteLastIcon}
                source={Images.delete_last}
              />
            </TouchableOpacity>
          </View>
          <View
            style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
          >
            {!this.props.isRecording ? (
              <TouchableOpacity
                onPress={() => {
                  this.props.onVideoFinishPressed();
                }}
              >
                <Image
                  style={portraitStyles.nextVideoIcon}
                  source={Images.next_video_icon}
                />
              </TouchableOpacity>
            ) : null}
          </View>

          <View style={{ flex: 1, alignItems: "center" }} />
        </View>
      );
    }
  }

  renderIOSLandscapeContainer() {
    return (
      <View
        style={{
          flex: 1,
          width: 90,
          marginTop: 10,
          position: "absolute",
          alignItems: "center",
          alignSelf: "flex-end",
          justifyContent: "center",
          height: "100%",
          backgroundColor: "#000",
          marginRight: 75,
        }}
      >
        <View
          style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
        >
          {!this.props.isRecording && !this.props.galleryPickerVisibility && (
            <TouchableOpacity
              style={{ marginTop: 10 }}
              onPress={() => {
                this.props.onPressSilhoutteMode();
              }}
            >
              <Image
                style={[
                  portraitStyles.pickFromGallery,
                  { height: 35, width: 35, resizeMode: "contain" },
                ]}
                source={
                  this.props.silhoutteSelected
                    ? Images.shadow_icon_selected
                    : Images.shadow_icon
                }
              />
            </TouchableOpacity>
          )}
        </View>
        <View style={{ flex: 1, justifyContent: "center" }}>
          {/* {this.props.hideCamera && ( */}
          <CameraCaptureAnimator
            ref={this.cameraCaptureAnimator}
            flash={this.props.flash}
            type={this.props.type}
            isVisible={!this.props.isVisible}
            orientation={this.props.orientationCheck}
            disableButton={this.props.disableButton}
            capturePressed={() => {
              this.props.capturePressed();
            }}
            captureReleased={() => {
              this.props.captureReleased();
            }}
            capturePhoto={() => {
              this.props.capturePhoto();
            }}
          />
          {/* )} */}
        </View>
        <View style={{ flex: 1, justifyContent: "center" }}>
          {!this.props.isRecording ? (
            this.renderLandscapeOptions()
          ) : (
            <View style={{ flex: 1, flexDirection: "row" }} />
          )}          
        </View>
      </View>
    );
  }

  render() {
    if (this.props.orientation == "portrait") {
      return this.renderIOSPortraitContainer();
    } else {
      return this.renderIOSLandscapeContainer();
    }
  }
}

const landscapeStyles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignSelf: "center",
    alignItems: "center",
    height: "100%",
    width: "100%",
    paddingLeft: utils.isIos ? 0 : 15,
    paddingRight: utils.isIos ? 0 : 15,
    position: "absolute",
    // zIndex:1,
    // right:0,
    // top:0,
    flex: 1,
    marginTop: utils.isIos ? 0 : 10,
  },
  captureLayoutLandscape: {
    flex: 1,
    height: "100%",
    position: "absolute",
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
  },
  pickFromGallery: {
    height: 24,
    width: 24,
  },
  nextVideoIcon: {
    height: 28,
    width: 28,
  },
  deleteLastIcon: {
    height: 32,
    width: 32,
  },
});

const portraitStyles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    bottom: 0,
    position: "absolute",
    paddingTop: 10,
    zIndex: 1,
  },
  captureLayoutPortrait: {
    marginTop: 10,
    flex: 1,
    width: "100%",
    minHeight: 80,
  },
  pickFromGallery: {
    height: 24,
    width: 24,
  },
  nextVideoIcon: {
    height: 32,
    width: 32,
    resizeMode:"contain",
    marginLeft:5
  },
  deleteLastIcon: {
    height: 32,
    width: 32,
  },
});
