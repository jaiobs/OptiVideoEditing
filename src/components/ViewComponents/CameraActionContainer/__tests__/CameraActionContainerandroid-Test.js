import React from 'react';
import renderer from 'react-test-renderer';
import CameraActionContainer from '../CameraActionContainer.android';

describe('CameraActionContainerTest', () => {
  it('should render the CameraActionContainer', async () => {
    jest.useFakeTimers();
    const tree = renderer.create(<CameraActionContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
