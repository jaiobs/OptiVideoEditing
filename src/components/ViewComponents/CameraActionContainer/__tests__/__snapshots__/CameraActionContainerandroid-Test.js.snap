// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraActionContainerTest should render the CameraActionContainer 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-end",
      "backgroundColor": "transparent",
      "flexDirection": "column",
      "height": "100%",
      "marginTop": 10,
      "position": "absolute",
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 40,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 4,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/res/assets/images/shadow_icon.png",
          }
        }
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <View>
    <View
      style={Object {}}
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingBottom": 26,
          "paddingTop": 8,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/res/assets/images/delete_last.png",
            }
          }
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
